Lib/dataclasses.py

```diff
@@ -808,8 +808,9 @@ def _get_field(cls, a_name, a_type, default_kw_only):
             raise TypeError(f'field {f.name} is a ClassVar but specifies '
                             'kw_only')
 
-    # For real fields, disallow mutable defaults for known types.
-    if f._field_type is _FIELD and isinstance(f.default, (list, dict, set)):
+    # For real fields, disallow mutable defaults.  Use unhashable as a proxy
+    # indicator for mutability.
+    if f._field_type is _FIELD and f.default.__hash__ is None:
         raise ValueError(f'mutable default {type(f.default)} for field '
```

Comment:
This test isn't correct. It needs to look at the class's `__hash__`, not the instance's.